gems.py

```diff
@@ -36,8 +36,11 @@ def __init__(self, gem_dict: dict, character_stats: 'Stats', socket: Optional[in
 		self.additional_effects = []
 		self.character_stats = character_stats
 		gem_data = self.get_gem_data()
-		self.tags = {gem_tag.lower() for gem_tag in (gem_data['tags'] or gem_data.get('types', []))}
-
+		self.tags = set()
+		if gem_data["tags"]:
+			self.tags = {tag.lower() for tag in gem_data["tags"]}
+		if gem_data.get("active_skill", {}).get("types"):
+			self.tags |= {tag.lower() for tag in gem_data["active_skill"]["types"]}
```

Comment:
```suggestion
		self.tags = {tag.lower() for tag in gem_data['tags']}
		self.tags |= {tag.lower() for tag in gem_data.get('active_skill', {}).get('types', [])}
```

Reply:
yeah i thought of something like this too,  but we need to check if the tags/types are None before iterating